### PR TITLE
phd2: init at 2.6.9dev1

### PR DESCRIPTION
**Pull request by hjones2199**
###### Motivation for this change
Add phd2 telescope guidance software to nixpkgs

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/astronomy/phd2/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchFromGitHub {
+    owner = "OpenPHDGuiding";
+    repo = "phd2";
+    rev = "v${version}";
+    sha256 = "1ih7m9lilh12xbhmwm9kkicaqy72mi3firl6df7m5x38n2zj3zm4";
+  };
+
+  nativeBuildInputs = [ pkgconfig wrapGAppsHook ];
+  buildInputs = [ cmake gtk3 wxGTK30-gtk3 curl gettext glib indilib libnova ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/astronomy/phd2/default.nix:16]: cmake should belong into nativeBuildInputs, too.

File: pkgs/applications/science/astronomy/phd2/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  enableParallelBuilding = true;
+
+  meta = {
+    description = "Telescope auto-guidance application";
+    maintainers = [ stdenv.lib.maintainers.hjones2199 ];
+    platforms = [ "x86_64-linux" ];
+    license = stdenv.lib.licenses.bsd3;
+    homepage = "https://openphdguiding.org/";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/science/astronomy/phd2/default.nix:30]:
```suggestion
  meta = with stdenv.lib; {
    homepage = "https://openphdguiding.org/";
    description = "Telescope auto-guidance application";
    license = licenses.bsd3;
    maintainers = with maintainers; [ hjones2199 ];
    platforms = [ "x86_64-linux" ];
  };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102200` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>phd2</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102200` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>phd2</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102200` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>phd2</li>
  </ul>
</details>

**Comment by hjones2199** [comment]
Thanks for the quick review. I pushed in the changes to meta, and moved cmake from buildInputs to nativeBuildInputs.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -3487,6 +3487,12 @@
     email = "t@larkery.com";
     name = "Tom Hinton";
   };
+  hjones2199 = {
+    email = "hjones2199@gmail.com";
+    github = "hjones2199";
+    githubId = 5525217;
+    name = "Hunter Jones";
+  };
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3495]: Please put this into a separate commit with the message ``maintainers: add hjones2199``.

File: pkgs/applications/science/astronomy/phd2/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  version = "2.6.9dev1";
+
+  src = fetchFromGitHub {
+    owner = "OpenPHDGuiding";
+    repo = "phd2";
+    rev = "v${version}";
+    sha256 = "1ih7m9lilh12xbhmwm9kkicaqy72mi3firl6df7m5x38n2zj3zm4";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig wrapGAppsHook ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/astronomy/phd2/default.nix:15]:
```suggestion
  nativeBuildInputs = [ cmake pkg-config wrapGAppsHook ];
```

File: pkgs/applications/science/astronomy/phd2/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{ stdenv, fetchFromGitHub, pkgconfig, cmake, gtk3, wrapGAppsHook,
```
Comment (SuperSandro2000) [inline pkgs/applications/science/astronomy/phd2/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, pkg-config, cmake, gtk3, wrapGAppsHook,
```

File: pkgs/applications/science/astronomy/phd2/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig wrapGAppsHook ];
+  buildInputs = [ gtk3 wxGTK30-gtk3 curl gettext glib indilib libnova ];
+
+  cmakeFlags = [
+    "-DOPENSOURCE_ONLY=1"
+  ];
+
+  enableParallelBuilding = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/astronomy/phd2/default.nix:22]: cmake enables that automatically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (hjones2199) [reply maintainers/maintainer-list.nix]: Just rebased onto master, since I added myself to the maintainer list in a different pull request this issue is solved. Working on your other suggestions as well.

File: pkgs/applications/science/astronomy/phd2/default.nix (hunk 4 above)
Comment (hjones2199) [reply pkgs/applications/science/astronomy/phd2/default.nix]: Changed in accordance with recommendation.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102200` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>phd2</li>
  </ul>
</details>
